### PR TITLE
[BugFix] Enhanced LifeCycleEvents in Android

### PR DESCRIPTION
**Pull request by 1abhishekpandey**
## Description of the change

- Build version was not saving when LifeCycleEvents is set to false.
- When recordScreenView is set to true and trackLifeCycleEvent is set to false then lifeCycleEvents was flowing. Now if this condition occurs then only screen event will be sent.
- [Notion Card](https://www.notion.so/rudderstacks/d5d15ce4be354e3caae11d4ea9f41477?v=8081ceba15644c85ad7a2cfb918a9c51&p=03c96e5431a045419c4755e410a15aa7)

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-android/107)

### CODE REVIEW

**Review by pallabmaiti — APPROVED** [review]

File: core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java [hunk 1]
```diff
@@ -195,11 +192,19 @@ private void checkApplicationUpdateStatus(Application application) {
...
             if (previousVersionCode == -1) {
-                // application was not installed previously, Application Installed events
-                RudderLogger.logDebug("Tracking Application Installed");
+                // application was not installed previously, Application Installed event
                 preferenceManager.saveBuildVersionCode(versionCode);
+                // If trackLifeCycleEvents is not allowed then discard the event
+                if (!isLifeCycleEventsAllowed) {
+                    return;
+                }
+                RudderLogger.logDebug("Tracking Application Installed");
```
Comment (desusai7) [inline core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java:207]: Looks like `Application Installed` and `Application Updated`, will be sent even though `trackLifeCycleEvents` is set  to false, can you verify that once ?

File: core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java [hunk 2]
```diff
@@ -195,11 +192,19 @@ private void checkApplicationUpdateStatus(Application application) {
...
+                application.registerActivityLifecycleCallbacks(this);
+            }
 
             if (previousVersionCode == -1) {
-                // application was not installed previously, Application Installed events
-                RudderLogger.logDebug("Tracking Application Installed");
+                // application was not installed previously, Application Installed event
                 preferenceManager.saveBuildVersionCode(versionCode);
+                // If trackLifeCycleEvents is not allowed then discard the event
+                if (!isLifeCycleEventsAllowed) {
```
Comment (desusai7) [inline core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java:204]: Can we also check if user opted out before sending `Application Installed` event as well ?

Comment (1abhishekpandey) [reply core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java:204]: Application Installed event will be triggered even before `optOut()` can be called. As optOut() feature depend upon EventRepository object. Please let me know, if I need to check more on this.

Comment (desusai7) [reply core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java:204]: Okay got it

File: core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java (hunk 1 above)
Comment (1abhishekpandey) [reply core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java:207]: It's working fine. When `trackLifeCycleEvents` is set to `false`, I cannot see any lifeCycleEvents (including `Application Installed` and `Application Updated` events) both in the `logcat` and `gw_jobs_1`. When it is set to `true`, lifeCycleEvents is flowing well. Please let me know if this solves the queries.

Comment (desusai7) [reply core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java:207]: Can you also try sett `recordScreenViews` to true at the same time ?

Comment (1abhishekpandey) [reply core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java:207]: Found the bug: When RecordScreenView is is set to true then also Application Installed event will be called. Fixing that.

**Review by desusai7 — APPROVED** [review]

**Review by pallabmaiti — APPROVED** [review]

**Pull request merged**